package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -24,85 +51,92 @@ def __init__(self, filename, **kwargs):
+        # will add more data 
 
         # set frame box dimensions
-        self.ts.dimensions = self._file['particles']['atoms']['box']['edges']['value'][frame,:,:] #triclinic dimension
-        #for i in range(3,6) :
-            #self.ts.dimensions[i] = np.arccos(self.ts.dimensions[i]) * 180.0 / np.pi
+        # set triclinic box vectors
+        ts._unitcell[:] = pyh5md.element(self._trajectory['box'], 'edges').value[frame, :]
 
         # set particle positions
-        frame_positions = self._file['particles']['atoms']['positions']['value'][frame,:,:]
+        frame_positions = pyh5md.element(self._trajectory, 'positions').value[frame, :]
         n_atoms_now = frame_positions.shape[0]
         if n_atoms_now != self.n_atoms :
             raise ValueError("Frame %d has %d atoms but the initial frame has %d"
                 " atoms. MDAnalysis in unable to deal with variable"
                 " topology!"%(frame, n_atoms_now, self.n_atoms))
         else :
-            self.ts.positions = frame_positions
+            ts.positions = frame_positions
             
         # set particle velocities
-        frame_velocities = self._file['particles']['atoms']['velocities']['value'][frame,:,:]
+        frame_velocities = pyh5md.element(self._trajectory, 'velocities').value[frame, :]
         n_atoms_now = frame_velocities.shape[0]
         if n_atoms_now != self.n_atoms :
             raise ValueError("Frame %d has %d atoms but the initial frame has %d"
                 " atoms. MDAnalysis in unable to deal with variable"
                 " topology!"%(frame, n_atoms_now, self.n_atoms))
         else :
-            self.ts.velocities = frame_velocities
+            ts.velocities = frame_velocities
             
         # set particle forces
-        frame_forces = self._file['particles']['atoms']['forces']['value'][frame,:,:]
+        frame_forces = pyh5md.element(self._trajectory, 'forces').value[frame, :]
         n_atoms_now = frame_forces.shape[0]
         if n_atoms_now != self.n_atoms :
             raise ValueError("Frame %d has %d atoms but the initial frame has %d"
                 " atoms. MDAnalysis in unable to deal with variable"
                 " topology!"%(frame, n_atoms_now, self.n_atoms))
         else :
-            self.ts.forces = frame_forces
+            ts.forces = frame_forces
             
         # set particle masses
-        frame_masses = self._file['particles']['atoms']['masses'][()]
+        frame_masses = pyh5md.element(self._trajectory, 'masses').value[()]
         n_atoms_now = frame_masses.shape
         if n_atoms_now != self.n_atoms :
             raise ValueError("Frame %d has %d atoms but the initial frame has %d"
                 " atoms. MDAnalysis in unable to deal with variable"
                 " topology!"%(frame, n_atoms_now, self.n_atoms))
         else :
-            self.ts.masses = frame_masses
+            ts.masses = frame_masses
```

Comment:
Timesteps do not have a `masses` attribute. MDAnalysis has no concept of particles changing mass over time. Leave this out for the moment.

We can later write a "Parser", something that reads the topology (atom names, masses, bonds (if available).

In your documentation, just state that "masses" are currently not read from a H5MD file – documenting missing features is a totally valid approach to not doing too much work in the beginning. It also helps you to get a working prototype out quickly. It also gives you well-defined tasks for improvement cycles later.